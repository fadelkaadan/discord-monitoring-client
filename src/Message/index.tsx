import styled from "styled-components";

interface MessageProps {
  content: String;
}

const Wrapper = styled.div`
  width: 500px;
  padding: 1rem 2rem;
  background-color: #5865f2;
  color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 3px;
`;

const Button = styled.button`
  border: none;
  outline: none;
  background-color: #de4141;
  color: #fff;
  padding: 5px;
  height: 30px;
  width: 30px;
  border-radius: 50%;
`;

const Message = ({ content }: MessageProps) => {
  return (
    <Wrapper>
      <p>{content}</p>
      <Button>x</Button>
    </Wrapper>
  );
};

export default Message;
