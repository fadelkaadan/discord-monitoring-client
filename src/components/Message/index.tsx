import styled from "styled-components";

interface MessageProps {
  content: String;
  onClick?: () => Promise<void>;
}

const Wrapper = styled.div`
  width: 500px;
  padding: 1rem 2rem;
  background-color: #5865f2;
  color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 3px;
`;

const Button = styled.button`
  border: none;
  outline: none;
  background-color: #de4141;
  color: #fff;
  padding: 5px;
  height: 30px;
  width: 30px;
  border-radius: 50%;

  :hover {
    background-color: #e44e4e;
    cursor: pointer;
  }
`;

const Message = ({ content, onClick }: MessageProps) => {
  return (
    <Wrapper>
      <p>{content}</p>
      <Button onClick={onClick}>x</Button>
    </Wrapper>
  );
};

export default Message;
