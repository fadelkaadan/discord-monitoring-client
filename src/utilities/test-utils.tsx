import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import "@testing-library/user-event";

interface AllTheProvidersProps {
  children?: any;
}

const AllTheProviders = ({ children }: AllTheProvidersProps) => {
  return <Router>{children}</Router>;
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
