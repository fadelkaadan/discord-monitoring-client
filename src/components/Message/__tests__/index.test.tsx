import { render, screen, RenderResult } from "../../../utilities/test-utils";

import Message from "../index";

describe("Message", () => {
  it("Should match the snapshot", () => {
    const { container }: RenderResult = render(
      <Message
        id="1"
        content="test content"
        handleIgnore={() => {}}
        handleDelete={() => {}}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Should render message's content", () => {
    const dummyContent = "test content";
    render(
      <Message
        id="1"
        content={dummyContent}
        handleIgnore={() => {}}
        handleDelete={() => {}}
      />
    );
    const content: HTMLElement = screen.getByText(dummyContent);
    expect(content.textContent).toEqual(dummyContent);
  });

  it("Should render two buttons", () => {
    const dummyContent = "test content";
    render(
      <Message
        id="1"
        content={dummyContent}
        handleIgnore={() => {}}
        handleDelete={() => {}}
      />
    );
    const buttons = screen.getAllByRole("button");
    expect(buttons.length).toEqual(1);
  });
});
