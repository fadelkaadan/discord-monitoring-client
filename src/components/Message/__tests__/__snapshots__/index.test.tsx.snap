// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Should match the snapshot 1`] = `
<div
  class="sc-bdnxRM gFNacz"
>
  <p
    class="sc-gtsrHT"
  >
    test content
  </p>
  <div
    class="sc-dlnjwi GCjvm"
  >
    <button
      class="sc-hKFxyN glSunK"
    >
      Ignore
    </button>
    <button
      class="sc-hKFxyN sc-eCApnc glSunK jjkkFJ"
    >
      Delete
    </button>
  </div>
</div>
`;
